modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2345,144 +2381,200 @@ private Expression<?>[] _getSelectExpressions(
 
-				Column<DynamicObjectDefinitionTable, Long>
-					relatedObjectDefinitionColumn =
-						(Column<DynamicObjectDefinitionTable, Long>)
-							_objectFieldLocalService.getColumn(
-								relatedObjectDefinition.getObjectDefinitionId(),
-								relatedField.getName());
+				for (String oDataFilterString : oDataFilterStrings) {
+					predicate = predicate.and(
+						_filterPredicateFactory.create(
+							oDataFilterString,
+							relatedObjectDefinition.getObjectDefinitionId()));
+				}
 
-				predicate = relatedObjectDefinitionColumn.eq(
-					dynamicObjectDefinitionTable.getPrimaryKeyColumn());
+				selectExpressions.add(
+					DSLQueryFactoryUtil.scalarSubDSLQuery(
+						joinStep.where(predicate),
+						DynamicObjectDefinitionTable.getJavaClass(
+							objectField.getDBType()),
+						objectField.getName(),
+						DynamicObjectDefinitionTable.getSQLType(
+							objectField.getDBType())));
 			}
-			else if (Objects.equals(
-						objectRelationship.getType(),
-						ObjectRelationshipConstants.TYPE_MANY_TO_MANY)) {
-
-				DynamicObjectRelationshipMappingTable
-					dynamicObjectRelationshipMappingTable =
-						new DynamicObjectRelationshipMappingTable(
-							objectDefinition.getPKObjectFieldDBColumnName(),
-							relatedObjectDefinition.
-								getPKObjectFieldDBColumnName(),
-							objectRelationship.getDBTableName());
-
-				Column<DynamicObjectRelationshipMappingTable, Long>
-					primaryKeyColumn2 =
-						dynamicObjectRelationshipMappingTable.
-							getPrimaryKeyColumn2();
-
-				joinStep = joinStep.innerJoinON(
-					dynamicObjectRelationshipMappingTable,
-					primaryKeyColumn2.eq(
-						relatedDynamicObjectDefinitionTable.
-							getPrimaryKeyColumn()));
-
-				Column<DynamicObjectRelationshipMappingTable, Long>
-					primaryKeyColumn1 =
-						dynamicObjectRelationshipMappingTable.
-							getPrimaryKeyColumn1();
-
-				predicate = primaryKeyColumn1.eq(
-					dynamicObjectDefinitionTable.getPrimaryKeyColumn());
-			}
-
-			List<String> oDataFilterStrings = TransformUtil.transform(
-				(List<ObjectFilter>)objectFieldSettingsValues.get("filters"),
-				objectFilter -> {
-					if (StringUtil.equals(
-							objectFilter.getFilterType(),
-							ObjectFilterConstants.TYPE_CURRENT_USER)) {
-
-						objectFilter.setJSON(
-							JSONUtil.put(
-								"currentUserId",
-								PrincipalThreadLocal.getUserId()
-							).toString());
+			else if (objectField.compareBusinessType(
+						ObjectFieldConstants.BUSINESS_TYPE_FORMULA)) {
+
+				List<ObjectField> objectFields =
+					_objectFieldLocalService.getObjectFields(
+						objectDefinition.getObjectDefinitionId());
+
+				Map<String, Object> columnMap = new HashMap<>();
+
+				for (ObjectField objectField2 : objectFields) {
+					if (Objects.equals(
+							objectField2.getBusinessType(),
+							ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT) ||
+						Objects.equals(
+							objectField2.getBusinessType(),
+							ObjectFieldConstants.BUSINESS_TYPE_RICH_TEXT)) {
```

Comment:
Here, you're avoiding only the attachment and the rich text fields.
What about the formula and aggregation fields? Is it desired regarding your logic?

Reply:
Good catch! I will check what happens with them. Ideally I should skip all the fields that don't have a corresponding column (virtual fields).